### PR TITLE
✨ feat: support GitHub OAuth 

### PR DESCRIPTION
**Pull request by devcxl**
#### 💻 变更类型 | Change Type

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] ⚡️ perf
- [ ] 📝 docs

#### 🔀 变更说明 | Description of Change

添加Github OAuth 登陆方式

#### 📝 补充信息 | Additional Information

新增环境变量：`GITHUB_CLIENT_ID` `GITHUB_CLIENT_SECRET`

### CODE REVIEW

File: README.zh-CN.md
Comment (arvinxx) [inline README.zh-CN.md:1]: 这个可以revert

**Review by arvinxx — CHANGES_REQUESTED** [review]
请参考其他几个PR补充下文档：
- https://github.com/lobehub/lobe-chat/pull/1456
- https://github.com/lobehub/lobe-chat/pull/1143

File: README.zh-CN.md
Comment (devcxl) [reply README.zh-CN.md:1]: OK

**Comment by devcxl** [comment]
@arvinxx 文档已补充

**Comment by arvinxx** [comment]
@devcxl OK，录个 demo 看一眼？

**Comment by devcxl** [comment]
好了，看这个
https://github.com/lobehub/lobe-chat/assets/64475363/9edbe3af-349d-4de0-8efb-e721e6549000

**Review by arvinxx — APPROVED** [review]
LGTM, I'll merge it. Thanks for your contribution!

**Comment by devcxl** [comment]
OK

**Pull request merged**